interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -135,16 +135,54 @@ namespace {
 namespace cling {
 
   Value::Value(const Value& other):
-    m_Storage(other.m_Storage), m_StorageType(other.m_StorageType),
+    m_Storage(other.m_Storage), m_NeedsManagedAlloc(other.m_NeedsManagedAlloc),
+    m_TypeKind(other.m_TypeKind),
     m_Type(other.m_Type), m_Interpreter(other.m_Interpreter) {
     if (other.needsManagedAllocation())
       AllocatedValue::getFromPayload(m_Storage.m_Ptr)->Retain();
   }
 
+  static Value::TypeKind getCorrespondingTypeKind(clang::QualType QT,
+                                                  const clang::ASTContext &C) {
+    using namespace clang;
+    if (const auto *ET = dyn_cast<EnumType>(QT.getTypePtr()))
+      QT = ET->getDecl()->getIntegerType();
+
+    if (C.hasSameType(QT, C.VoidTy))
```

Comment:
```suggestion
    else if (C.hasSameType(QT, C.VoidTy))
```

Reply:
It's an optimization; `hasSameType` isn't cheap so at least avoid it for `EnumType`s?

Reply:
The implementation of `hasSameType` is:
```cpp
bool hasSameType(QualType T1, QualType T2) const {
    return getCanonicalType(T1) == getCanonicalType(T2);
}
```

This is quite often used in clang. Note we use it only when the value is being constructed and that is not on the critical path.